### PR TITLE
Remove space in default header

### PR DESCRIPTION
**Pull request by BrennanConroy**
@danroth27 @Tratcher 

Temporary change since cross-plat coreclr HttpClient has a bug

### CODE REVIEW

**Comment by danroth27** [comment]
:shipit:

**Comment by ChengTian** [comment]
:shipit:

**Comment by davidfowl** [comment]
After the ci passes :shipit:

**Comment by BrennanConroy** [comment]
Changed it a little since Mono didn't like our new header.

File: src/Microsoft.Dnx.Tooling/Restore/NuGet/HttpSource.cs [hunk 1]
```diff
@@ -70,8 +70,8 @@ internal class HttpSource
             }
 
             var env = RuntimeEnvironmentHelper.RuntimeEnvironment;
-            var userAgentHeader = $"{UserAgentName}/{env.RuntimeVersion} ({env.OperatingSystem} {env.OperatingSystemVersion})";
-            _client.DefaultRequestHeaders.UserAgent.ParseAdd(userAgentHeader);
+            var userAgentHeader = $"{UserAgentName}/{env.RuntimeVersion}({env.OperatingSystem}{env.OperatingSystemVersion})";
+            _client.DefaultRequestHeaders.TryAddWithoutValidation("user-agent", userAgentHeader);
```
Comment (ChengTian) [inline src/Microsoft.Dnx.Tooling/Restore/NuGet/HttpSource.cs]: RFC doesn't mention this is case-insensitive, let's make it `User-Agent`

Comment (ChengTian) [inline src/Microsoft.Dnx.Tooling/Restore/NuGet/HttpSource.cs]: Actually, it must be case-sensitive:
https://www.google.com/webhp?sourceid=chrome-instant&ion=1&espv=2&ie=UTF-8#q=http%20header%20case%20sensitive

Comment (ChengTian) [inline src/Microsoft.Dnx.Tooling/Restore/NuGet/HttpSource.cs]: My bad, it's case-insensitive. Ignore this comment

**Comment by ChengTian** [comment]
:shipit:

File: src/Microsoft.Dnx.Tooling/Restore/NuGet/HttpSource.cs (hunk 1 above)
Comment (davidfowl) [inline src/Microsoft.Dnx.Tooling/Restore/NuGet/HttpSource.cs]: Change it to User-Agent anyways

**Comment by davidfowl** [comment]
:shipit:

**Pull request merged**